### DISCUSSION TITLE
How to  check if a field(column)  exists in a Prisma model without making a database call

### DISCUSSION THREAD

**Discussion by ibmmt** [prisma/prisma#18349] [Q&A] [answered]
How to get all  fields.

**Comment by nurul3101** [comment]
Hello @ibrahimmt 👋 

I am not sure if it's possible to check without interacting with the database.
You can use `npx prisma migrate diff` command ([Reference](https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff)) to check for the difference between your database and the models defined in the schema file

**Reply by ludralph** [reply]
I think you can access the fields using `Prisma.dmmf.datamodel.model`
```js
const {  Prisma } = require("@prisma/client")
console.log(Prisma.dmmf.datamodel.models[0].fields)
````
You would then need to write logic to see if the field exist

**Reply by ibmmt** [reply]
@ludralph @nurul3101 Thank you for your answer 

I generated an Object from datamodel  ,  @ludralph answer
 
```
import { Prisma } from '@prisma/client';
const obj: any = {};
const prismaModels = Prisma.dmmf.datamodel.models;
for (const model of prismaModels) {
  const modelName = model.name as string;
  if (modelName) {
    obj[modelName] = {};
    for (const field of model.fields) {
      const fieldName = field.name as string;
      obj[modelName][fieldName] = true;
    }
  }
}
export const PrimsaOrmObject = obj;
```

**Comment by ibmmt** [comment, marked as answer]
@ludralph @nurul3101 Thank you for your answer

I generated an Object from datamodel , @ludralph answer
```

import { Prisma } from '@prisma/client';
const obj: any = {};
const prismaModels = Prisma.dmmf.datamodel.models;
for (const model of prismaModels) {
  const modelName = model.name as string;
  if (modelName) {
    obj[modelName] = {};
    for (const field of model.fields) {
      const fieldName = field.name as string;
      obj[modelName][fieldName] = true;
    }
  }
}
export const PrimsaOrmObject = obj;
```

**Reply by nurul3101** [reply]
Thanks for letting us know 👍
